File: code/ilmenau_deu_aut_che_historic.r
files_deu <- list.files("data-raw/Ilmenau DEU", pattern = "*deu_raw.csv")

for (f in files_deu) {
    # Germany
    input_filename <- paste("data-raw/Ilmenau DEU/", f, sep = "")
    input_dataframe <- read.csv(input_filename, sep = ",")
    output_dataframe <- data.frame()
    cnames <- c("data_version", "target", "date", "location", "type", "quantile", "value")
    output_filename <- paste("data-processed/ilmenau/",input_dataframe[nrow(input_dataframe), 1], "-ilmenau.csv", sep = "")

    # some files don't exist for all countries
    if ((substr(f, 1, 10) == "2021-01-27") || (substr(f, 1, 10) == "2021-03-02")) {
       next()
    }

    for (row in seq(1, nrow(input_dataframe))) {
        output_dataframe <- rbind(output_dataframe, list(input_dataframe[nrow(input_dataframe), 1], "7 day R", input_dataframe[row, 1], "DE", "point", "NA", input_dataframe[row, 3]))
        output_dataframe <- rbind(output_dataframe, list(input_dataframe[nrow(input_dataframe), 1], "7 day R", input_dataframe[row, 1], "DE", "quantile", "0.025", input_dataframe[row, 5]))
        output_dataframe <- rbind(output_dataframe, list(input_dataframe[nrow(input_dataframe), 1], "7 day R", input_dataframe[row, 1], "DE", "quantile", "0.975", input_dataframe[row, 6]))

    }
    # Austria
    input_filename <- paste("data-raw/Ilmenau AUT/", substr(f, 1, 10), "_ilmenau_aut_raw.csv", sep = "")
    input_dataframe <- read.csv(input_filename, sep = ",")
    # fist file has a different format
    if (substr(f, 1, 10) == "2020-11-02") {
        for (row in seq(1, nrow(input_dataframe))) {
        output_dataframe <- rbind(output_dataframe, list(input_dataframe[nrow(input_dataframe), 3], "7 day R", input_dataframe[row, 3], "AU", "point", "NA", input_dataframe[row, 12]))
        output_dataframe <- rbind(output_dataframe, list(input_dataframe[nrow(input_dataframe), 3], "7 day R", input_dataframe[row, 3], "AU", "quantile", "0.025", input_dataframe[row, 14]))
        output_dataframe <- rbind(output_dataframe, list(input_dataframe[nrow(input_dataframe), 3], "7 day R", input_dataframe[row, 3], "AU", "quantile", "0.975", input_dataframe[row, 15]))

        }
    } else {
        for (row in seq(1, nrow(input_dataframe))) {
        output_dataframe <- rbind(output_dataframe, list(input_dataframe[nrow(input_dataframe), 1], "7 day R", input_dataframe[row, 3], "AU", "point", "NA", input_dataframe[row, 3]))
        output_dataframe <- rbind(output_dataframe, list(input_dataframe[nrow(input_dataframe), 1], "7 day R", input_dataframe[row, 3], "AU", "quantile", "0.025", input_dataframe[row, 5]))
        output_dataframe <- rbind(output_dataframe, list(input_dataframe[nrow(input_dataframe), 1], "7 day R", input_dataframe[row, 3], "AU", "quantile", "0.975", input_dataframe[row, 6]))

        }
    }

    # Switzerland
    input_filename <- paste("data-raw/Ilmenau CHE/", substr(f, 1, 10), "_ilmenau_che_raw.csv", sep = "")
    input_dataframe <- read.csv(input_filename, sep = ",")
    # fist file has a different format
    if (substr(f, 1, 10) == "2020-11-02") {
        for (row in seq(1, nrow(input_dataframe))) {
        output_dataframe <- rbind(output_dataframe, list(input_dataframe[nrow(input_dataframe), 3], "7 day R", input_dataframe[row, 3], "CH", "point", "NA", input_dataframe[row, 12]))
        output_dataframe <- rbind(output_dataframe, list(input_dataframe[nrow(input_dataframe), 3], "7 day R", input_dataframe[row, 3], "CH", "quantile", "0.025", input_dataframe[row, 14]))
        output_dataframe <- rbind(output_dataframe, list(input_dataframe[nrow(input_dataframe), 3], "7 day R", input_dataframe[row, 3], "CH", "quantile", "0.975", input_dataframe[row, 15]))

        }
    } else {
        for (row in seq(1, nrow(input_dataframe))) {
        output_dataframe <- rbind(output_dataframe, list(input_dataframe[nrow(input_dataframe), 1], "7 day R", input_dataframe[row, 3], "CH", "point", "NA", input_dataframe[row, 3]))
        output_dataframe <- rbind(output_dataframe, list(input_dataframe[nrow(input_dataframe), 1], "7 day R", input_dataframe[row, 3], "CH", "quantile", "0.025", input_dataframe[row, 5]))
        output_dataframe <- rbind(output_dataframe, list(input_dataframe[nrow(input_dataframe), 1], "7 day R", input_dataframe[row, 3], "CH", "quantile", "0.975", input_dataframe[row, 6]))

        }
    }

    colnames(output_dataframe) <- cnames
    write.csv(output_dataframe, output_filename, row.names = FALSE)
}

File: code/ilmenau_deu_aut_che_historic.py
import pandas as pd
import os
from os.path import exists

path = "../data-raw/Ilmenau DEU/"
files = os.listdir(path)

for file in files:
    if( exists("../data-raw/Ilmenau AUT/" + file[0:10] + "_ilmenau_aut_raw.csv") and exists("../data-raw/Ilmenau CHE/" + file[0:10] + "_ilmenau_che_raw.csv")):
        data_deu = pd.read_csv(path+file)
        data_aut = pd.read_csv("../data-raw/Ilmenau AUT/" + file[0:10] + "_ilmenau_aut_raw.csv")
        data_che = pd.read_csv("../data-raw/Ilmenau CHE/" + file[0:10] + "_ilmenau_che_raw.csv")
        df = pd.DataFrame()

        for i in range(0, len(data_deu)):
            df = df.append(pd.DataFrame({'data_version': [file[0:10], file[0:10], file[0:10]], 'target': ["7 day R", "7 day R", "7 day R"], 'date': [data_deu.iloc[i, 0], data_deu.iloc[i, 0], data_deu.iloc[i, 0]], 'location': ["DE", "DE", "DE"], 'type': ["point", "quantile", "quantile"], 'quantile': ["NA", "0.025", "0.975"], 'value': [data_deu.iloc[i, 2], data_deu.iloc[i, 4], data_deu.iloc[i, 5]]}))
        
        for i in range(0, len(data_aut)):
            df = df.append(pd.DataFrame({'data_version': [file[0:10], file[0:10], file[0:10]], 'target': ["7 day R", "7 day R", "7 day R"], 'date': [data_aut.iloc[i, 0], data_aut.iloc[i, 0], data_aut.iloc[i, 0]], 'location': ["AT", "AT", "AT"], 'type': ["point", "quantile", "quantile"], 'quantile': ["NA", "0.025", "0.975"], 'value': [data_aut.iloc[i, 2], data_aut.iloc[i, 4], data_aut.iloc[i, 5]]}))
        
        for i in range(0, len(data_che)):
            df = df.append(pd.DataFrame({'data_version': [file[0:10], file[0:10], file[0:10]], 'target': ["7 day R", "7 day R", "7 day R"], 'date': [data_che.iloc[i, 0], data_che.iloc[i, 0], data_che.iloc[i, 0]], 'location': ["CH", "CH", "CH"], 'type': ["point", "quantile", "quantile"], 'quantile': ["NA", "0.025", "0.975"], 'value': [data_che.iloc[i, 2], data_che.iloc[i, 4], data_che.iloc[i, 5]]}))
        
        df.to_csv("../data-processed/ilmenau/" + file[0:10] + "-ilmenau.csv", index=False, index_label=False)
    
